src/arch/arm/object/smc.c

```diff
@@ -8,18 +8,17 @@
 #ifdef CONFIG_ALLOW_SMC_CALLS
 #include <arch/object/smc.h>
 
-static exception_t invokeSMCCall(word_t *buffer)
+compile_assert(n_msgRegisters_less_than_smc_regs, n_msgRegisters <= NUM_SMC_REGS);
+
+static exception_t invokeSMCCall(word_t *buffer, bool_t call)
 {
     word_t i;
     seL4_Word arg[NUM_SMC_REGS];
-    word_t *ipcBuffer;
 
     for (i = 0; i < NUM_SMC_REGS; i++) {
         arg[i] = getSyscallArg(i, buffer);
     }
 
-    ipcBuffer = lookupIPCBuffer(true, NODE_STATE(ksCurThread));
```

Comment:
I don't think you can safely remove this, however tempting:

 > Require the input length to be >= NUM_SMC_REGS. This ensures that any  accesses to the IPC buffer memory is safe (if the IPC buffer isn't  present then the message length will be truncated to n_msgRegisters).

This misses the corner case where the IPC buffer is mapped read-only.

Reply:
Yea you're right good catch. TIL that IPC buffer caps can be without write rights.  (seems like such an anti-feature)

Reply:
Added it back in